src/ir.cpp

```diff
@@ -26618,8 +26636,14 @@ static IrInstGen *ir_analyze_instruction_slice(IrAnalyze *ira, IrInstSrcSlice *i
     if (type_is_invalid(result_loc->value->type) || result_loc->value->type->id == ZigTypeIdUnreachable) {
         return result_loc;
     }
-    return ir_build_slice_gen(ira, &instruction->base.base, return_type,
-        ptr_ptr, casted_start, end, instruction->safety_check_on, result_loc);
+
+    if (generate_non_zero_assert) {
+        IrInstGen *c_ptr_val = ir_get_deref(ira, &instruction->base.base, ptr_ptr, nullptr);
```

Comment:
`s/c_ptr_val/ptr_val`, this code path is not specific for C pointers (even though atm we only trigger it for C pointers).
Some error checking is also good to have:
```zig
        if (type_is_invalid(ptr_val->value->type))
            return ira->codegen->invalid_inst_gen;
```